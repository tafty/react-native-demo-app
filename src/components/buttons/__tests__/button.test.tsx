import 'react-native'
import React from 'react'
import { ActivityIndicator } from 'react-native'
import { render, fireEvent } from 'react-native-testing-library'

import Button from '../button'

describe('Button', () => {
  it('should render enabled', () => {
    const isDisabled = false
    const onPress = jest.fn()
    const testID = 'test-id'
    const label = 'photos'

    const { getByTestId, toJSON } = render(
      <Button
        isDisabled={isDisabled}
        onPress={onPress}
        testID={testID}
        label={label}
      />,
    )

    expect(getByTestId(`${testID}-label`).props.children).toBe(
      label.toUpperCase(),
    )

    fireEvent.press(getByTestId(testID))
    expect(onPress).toBeCalled()

    expect(toJSON()).toMatchSnapshot()
  })

  it('should render disabled', () => {
    const isDisabled = true
    const onPress = jest.fn()
    const testID = 'test-id'
    const label = 'photos'

    const { getByTestId, toJSON } = render(
      <Button
        isDisabled={isDisabled}
        onPress={onPress}
        testID={testID}
        label={label}
      />,
    )

    expect(getByTestId(`${testID}-label`).props.children).toBe(
      label.toUpperCase(),
    )

    expect(toJSON()).toMatchSnapshot()
  })

  it('should render enabled with an activity indicator', () => {
    const isDisabled = false
    const onPress = jest.fn()
    const testID = 'test-id'
    const label = 'photos'

    const { getByTestId, toJSON } = render(
      <Button
        isDisabled={isDisabled}
        onPress={onPress}
        showActivityIndicator={true}
        testID={testID}
        label={label}
      />,
    )

    const style = { color: '#343a40', flex: 0, marginRight: 8 }

    const indicator = (
      <ActivityIndicator
        animating={true}
        color="#999999"
        hidesWhenStopped={true}
        size="small"
        style={style}
      />
    )

    expect(getByTestId(`${testID}-label`).props.children.toString()).toBe(
      indicator.toString(),
    )

    fireEvent.press(getByTestId(testID))
    expect(onPress).toBeCalled()

    expect(toJSON()).toMatchSnapshot()
  })

  it('should render disabled with an activity indicator', () => {
    const isDisabled = true
    const onPress = jest.fn()
    const testID = 'test-id'
    const label = 'photos'

    const { getByTestId, toJSON } = render(
      <Button
        isDisabled={isDisabled}
        onPress={onPress}
        showActivityIndicator={true}
        testID={testID}
        label={label}
      />,
    )

    const style = { color: '#343a40', flex: 0, marginRight: 8 }

    const indicator = (
      <ActivityIndicator
        animating={true}
        color="#999999"
        hidesWhenStopped={true}
        size="small"
        style={style}
      />
    )

    expect(getByTestId(`${testID}-label`).props.children.toString()).toBe(
      indicator.toString(),
    )

    expect(toJSON()).toMatchSnapshot()
  })
})
