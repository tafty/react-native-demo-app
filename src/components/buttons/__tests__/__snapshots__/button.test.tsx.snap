// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render disabled 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(208, 208, 208, 1)",
        "borderColor": "rgba(128, 128, 128, 1)",
        "borderRadius": 4,
        "borderWidth": 1,
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "padding": 16,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "rgba(128, 128, 128, 1)",
          "flex": 0,
          "fontSize": 16,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
      testID="test-id-label"
    >
      PHOTOS
    </Text>
  </View>
</View>
`;

exports[`Button should render disabled with an activity indicator 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(208, 208, 208, 1)",
        "borderColor": "rgba(128, 128, 128, 1)",
        "borderRadius": 4,
        "borderWidth": 1,
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "padding": 16,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "rgba(128, 128, 128, 1)",
          "flex": 0,
          "fontSize": 16,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
      testID="test-id-label"
    >
      <ActivityIndicator
        animating={true}
        color="#999999"
        hidesWhenStopped={true}
        size="small"
        style={
          Object {
            "color": "rgba(128, 128, 128, 1)",
            "flex": 0,
            "marginRight": 8,
          }
        }
      />
    </Text>
  </View>
</View>
`;

exports[`Button should render enabled 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 124, 255, 1)",
        "borderColor": "rgba(0, 124, 255, 1)",
        "borderRadius": 4,
        "borderWidth": 1,
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "padding": 16,
      }
    }
    testID="test-id"
  >
    <Text
      style={
        Object {
          "color": "rgba(255, 255, 255, 1)",
          "flex": 0,
          "fontSize": 16,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
      testID="test-id-label"
    >
      PHOTOS
    </Text>
  </View>
</View>
`;

exports[`Button should render enabled with an activity indicator 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 124, 255, 1)",
        "borderColor": "rgba(0, 124, 255, 1)",
        "borderRadius": 4,
        "borderWidth": 1,
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "padding": 16,
      }
    }
    testID="test-id"
  >
    <Text
      style={
        Object {
          "color": "rgba(255, 255, 255, 1)",
          "flex": 0,
          "fontSize": 16,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
      testID="test-id-label"
    >
      <ActivityIndicator
        animating={true}
        color="#999999"
        hidesWhenStopped={true}
        size="small"
        style={
          Object {
            "color": "rgba(0, 124, 255, 1)",
            "flex": 0,
            "marginRight": 8,
          }
        }
      />
    </Text>
  </View>
</View>
`;
